src/OpenIddict.EF/OpenIddictContext.cs

```diff
@@ -10,29 +10,120 @@
 using OpenIddict.Models;
 
 namespace OpenIddict {
-    public class OpenIddictContext<TUser, TApplication, TRole, TKey> : IdentityDbContext<TUser, TRole, TKey>
+    /// <summary>
+    /// Base class for the Entity Framework database context used for OpenIddict.
+    /// </summary>
+    /// <typeparam name="TApplication">The type of application objects.</typeparam>
+    /// <typeparam name="TApplicationKey">The type used for the primary key for the application.</typeparam>
+    /// <typeparam name="TUser">The type of user objects.</typeparam>
+    /// <typeparam name="TRole">The type of role objects.</typeparam>
+    /// <typeparam name="TKey">The type of the primary key for users and roles.</typeparam>
+    /// <seealso cref="Microsoft.AspNetCore.Identity.EntityFrameworkCore.IdentityDbContext{TUser, TRole, TKey}" />
+    public class OpenIddictContext<TApplication, TApplicationKey, TUser, TRole, TKey> : IdentityDbContext<TUser, TRole, TKey>
```

Comment:
No, I mean that if you want to have a key type specific to the `Application` entity, then someone else will want a key type for `Authorization`, another one for `Scope` and a last one for `Token`.

We can't afford having a key type for each entity.